sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_client_base_async.py

```diff
@@ -194,78 +193,88 @@ def _create_auth(self):
-        :type consumer_group: str
-        :param partition_id: The identifier of the Event Hub partition from which events will be received.
-        :type partition_id: str
-        :param event_position: The position within the partition where the consumer should begin reading events.
-        :type event_position: ~azure.eventhub.common.EventPosition
-        :param owner_level: The priority of the exclusive consumer. The client will create an exclusive
-         consumer if owner_level is set.
-        :type owner_level: int
-        :param prefetch: The message prefetch count of the consumer. Default is 300.
-        :type prefetch: int
-        :param track_last_enqueued_event_properties: Indicates whether or not the consumer should request information
-         on the last enqueued event on its associated partition, and track that information as events are received.
-         When information about the partition's last enqueued event is being tracked, each event received from the
-         Event Hubs service will carry metadata about the partition. This results in a small amount of additional
-         network bandwidth consumption that is generally a favorable trade-off when considered against periodically
-         making requests for partition properties using the Event Hub client.
-         It is set to `False` by default.
-        :type track_last_enqueued_event_properties: bool
-        :param loop: An event loop. If not specified the default event loop will be used.
-        :rtype: ~azure.eventhub.aio.consumer_async.EventHubConsumer
-        """
-        owner_level = kwargs.get("owner_level")
-        prefetch = kwargs.get("prefetch") or self._config.prefetch
-        track_last_enqueued_event_properties = kwargs.get("track_last_enqueued_event_properties", False)
-        loop = kwargs.get("loop")
-
-        source_url = "amqps://{}{}/ConsumerGroups/{}/Partitions/{}".format(
-            self._address.hostname, self._address.path, consumer_group, partition_id)
-        handler = EventHubConsumer(
-            self, source_url, event_position=event_position, owner_level=owner_level,
-            prefetch=prefetch,
-            track_last_enqueued_event_properties=track_last_enqueued_event_properties, loop=loop)
-        return handler
-
-    def _create_producer(
-            self, *,
-            partition_id: str = None,
-            send_timeout: float = None,
-            loop: asyncio.AbstractEventLoop = None
-    ) -> EventHubProducer:
+    async def close(self):
+        # type: () -> None
+        await self._conn_manager.close_connection()
+
+
+class ConsumerProducerMixin(object):
+
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.close()
+
+    def _check_closed(self):
+        if self.closed:
+            raise EventHubError("{} has been closed. Please create a new one to handle event data.".format(self._name))
```

Comment:
This may warrant its own exception type.

Reply:
I don't think this actually gets propagated up to users. But will do so anyway.